lib/src/js/interface.dart

```diff
@@ -0,0 +1,37 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/// Whether we're running as JS (browser or Node.js).
+const bool isJS = false;
+
+/// Whether we're running as Node.js (not browser or Dart VM).
+bool get isNodeJs => throw '';
+
+/// Whether we're running as browser (not Node.js or Dart VM).
```

Comment:
This isn't an accurate definition of a browser, and that's misleading. If we're going to provide this getter, it should check for the existence of a DOM API like `window.href`.

Reply:
I'm assuming you meant `location.href`, interestingly the node runs in `npm_test.dart` actually mocks `location = { get href() => ... }` and returns a `file:///` path o.O

So.. I'll use `history.scrollRestoration` instead as the signal for being a browser